### PR TITLE
Enhancement: Better support null and undefined values in useRouteQueryParam

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Adds default support for undefined params and adds nullable versions of all the formatters. 

I also added a date formatter since they're wasn't one already.

### CODE REVIEW

File: src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { LocationQueryValue } from 'vue-router'
+import { InvalidRouteParamValue } from '@/useRouteQueryParam/formats/InvalidRouteParamValue'
+import { RouteParam } from '@/useRouteQueryParam/formats/RouteParam'
+
+export class NullableBooleanRouteParam extends RouteParam<boolean | null> {
+
+  protected override parse(value: LocationQueryValue): boolean | null {
```
Comment (stackoverfloweth) [inline src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts:7]: how does this ever return `null` ?

File: src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if (value === 'true') {
+      return true
+    }
+
+    if (value === 'false') {
+      return false
+    }
```
Comment (stackoverfloweth) [inline src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts:14]: any reasonable way to reuse the `BooleanRouteParam` class here?

**Review by stackoverfloweth — APPROVED** [review]

File: src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts (hunk 2 above)
Comment (pleek91) [reply src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts:14]: Not without refactoring something. `BooleanRouteParam` extends `RouteParam<boolean>` which is what sets the types for the `parse` and `format` methods. Extending `BooleanRouteParam` and returning `null` throws ts errors. 

Very open to ideas here. But didn't think it was worth putting to much effort into reusing a few lines of code.

File: src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts (hunk 1 above)
Comment (pleek91) [reply src/useRouteQueryParam/formats/NullableBooleanRouteParam.ts:7]: id didn't. fixed! Good catch.

**Pull request merged**